maro/rl/agent/abs_agent_manager.py

```diff
@@ -71,12 +76,27 @@ def __getitem__(self, agent_id):
         return self._agent_dict[agent_id]
 
     def _assemble(self, agent_dict):
-        """
-        abstract method to populate the _agent_dict attribute.
+        """Assembles agents and fill the `agent_dict` with them.
         """
         return NotImplemented
 
     def choose_action(self, decision_event, snapshot_list):
+        """This is the interface for interacting with the environment.
+
+        The method consists of 4 steps:
+        1. The decision event and snapshot list are converted by the state shaper to a model input. The state shaper
+           also finds the target agent ID.
+        2. The target agent takes the model input and uses its underlying models to compute an action.
+        3. Key information regarding the transition is recorded in the `_trajectory` attribute.
+        4. The action computed by the model is converted to an environment executable action by the action shaper.
```

Comment:
double \`,  words with single \`  only shows as italics in APIDOC. Same to other key words